Closes #327 -- ignore cases where order changes during diff process

zfisch opened the pull request:
It turns out we only delimit using a semicolon on the `Pathogenicity_all` field, which means the comma delimited search we already have in place should catch other reorders.

Given that, I found that simply checking to make sure the exact same characters are in both the concatenated new and old values for `Pathogenicity_all` is a reliable check.

zfisch commented:
Just realized a potential issue if classifications swapped between sources (e.g. `Benign (ClinVar); Pathogenic (BIC)` -> `Pathogenic (ClinVar); Benign (BIC)`) would not be caught. Update coming soon!

zfisch commented:
Updated to address the aforementioned issue and add unit tests.

melissacline commented:
> add unit tests
w00t!!!

On Tue, Dec 27, 2016 at 5:55 PM, Zack Fischmann <notifications@github.com>
wrote:

> Updated to address the aforementioned issue and add unit tests.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BD2KGenomics/brca-exchange/pull/340#issuecomment-269412145>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABoqh48HF2zXY5WtmAwfQnkA1PPinZdJks5rMcGggaJpZM4LWehP>
> .
>

zfisch commented:
@melissacline unfortunately the issue is slightly more complex than splitting on semicolons and then splitting on commas (e.g. if we split `Uncertain_significance,Likely_benign,not_provided (ClinVar); Pending (BIC) | not_provided,Likely_benign,Uncertain_significance (ClinVar); Pending (BIC)` on semicolons and then commas, it would return a false negative of `not_provided (ClinVar) != Uncertain_significance (ClinVar)` even though the original values are in fact equivalent. This is the reason I switched to a character comparison instead.

I did make an update, however, to use the sorted comparison in this PR for all semicolon delimited values instead of only `Pathogenicity_all`. That being said, only `Pathogenicity_all` and `Condition_ID_value_ENIGMA` have semicolons.

zfisch commented:
Ack, it actually requires special treatment with or without semicolons. As such, I renamed the function to clarify that it is an edge case particular to `Pathogenicity_all`.
